src/terraform-provider-paperspace/client.go

```diff
@@ -60,22 +60,17 @@ func logHttpRequestConstruction(operationType string, url string, data *bytes.Bu
 	log.Printf("Constructing %s request to url: %s, data: %v", operationType, url, data)
 }
 
-// logHttpResponseObject logs http response fields
-func logHttpResponseObject(reqURL *url.URL, resp *http.Response, body map[string]interface{}, err error) {
-	log.Printf("Request URL: %v", reqURL)
-	log.Printf("Response Status: %v", resp.Status)
-	log.Printf("Response: %v", resp)
-	log.Printf("Response Body: %s", body)
-	log.Printf("Error: %v", err)
-}
-
-// LogArrayResponse logs http response fields
-func LogHttpResponseArray(reqDesc string, reqURL *url.URL, resp *http.Response, body interface{}, err error) {
+// LogHttpResponse logs http response fields
+func LogHttpResponse(reqDesc string, reqURL *url.URL, resp *http.Response, body interface{}, err error) {
+	jsonBody, err2 := json.Marshal(body)
+	if err2 != nil {
+		log.Printf("Error unmarshaling response body: %v", err2)
+	}
 	log.Printf("Request: %v", reqDesc)
 	log.Printf("Request URL: %v", reqURL)
 	log.Printf("Response Status: %v", resp.Status)
 	log.Printf("Response: %v", resp)
-	log.Printf("Response Body: %s", body)
+	log.Printf("Response Body: %s", jsonBody)
```

Comment:
Instead of re serializing to json, I'd recommend using `spew.SDump` which can pretty print structs including those with pointers. https://godoc.org/github.com/davecgh/go-spew/spew#Sdump

Reply:
ah ok cool -- i'll use that -- thanks